kafka/metrics.go

```diff
@@ -311,70 +311,78 @@ func formatMetricError(name string, err error) error {
 }
 
 func (h *metricHooks) OnBrokerConnect(meta kgo.BrokerMetadata, _ time.Duration, _ net.Conn, err error) {
-	attributes := attribute.NewSet(
-		semconv.MessagingSystem("kafka"),
-	)
+	attrs := make([]attribute.KeyValue, 0, 2)
+	attrs = append(attrs, semconv.MessagingSystem("kafka"))
+	if h.namespace != "" {
+		attrs = append(attrs, attribute.String("namespace", h.namespace))
+	}
```

Comment:
It seems like this could be extracted into a function to avoid the 4x repetition. Up to you.